pkg/catalogv2/system/system.go

```diff
@@ -257,52 +259,76 @@ func podDone(chart string, newPod *corev1.Pod) (bool, error) {
 	return false, nil
 }
 
-func (m *Manager) isInstalled(namespace, name, version, minVersion string, values map[string]interface{}) (bool, error) {
+// isInstalled returns whether the release is installed, if false, it will return the version and values.yaml it should install/upgrade
+func (m *Manager) isInstalled(namespace, name, version, minVersion string, desiredValue map[string]interface{}) (bool, string, map[string]interface{}, error) {
 	helmcfg := &action.Configuration{}
 	if err := helmcfg.Init(m.restClientGetter, namespace, "", logrus.Infof); err != nil {
-		return false, err
+		return false, "", nil, err
 	}
 
 	l := action.NewList(helmcfg)
 	l.Filter = "^" + name + "$"
 
 	releases, err := l.Run()
 	if err != nil {
-		return false, err
+		return false, "", nil, err
 	}
 
 	desired, err := semver.NewVersion(version)
 	if err != nil {
-		return false, err
+		return false, "", nil, err
 	}
 
 	for _, release := range releases {
 		if release.Info.Status != release2.StatusDeployed {
 			continue
 		}
 
+		desiredValuesJSON, err := json.Marshal(desiredValue)
+		if err != nil {
+			return false, "", nil, err
+		}
+
+		actualValueJSON, err := json.Marshal(release.Config)
+		if err != nil {
+			return false, "", nil, err
+		}
+
+		patchedJSON, err := jsonpatch.MergePatch(actualValueJSON, desiredValuesJSON)
+		if err != nil {
+			return false, "", nil, err
+		}
+
+		if err := json.Unmarshal(patchedJSON, &desiredValue); err != nil {
```

Comment:
You need to do `desiredValue = map[string]interface{}` otherwise you might be aggregating expected values.